Automattic/jetpack
# Sync: Disable custom table when error in db stating the  jetpack_sync_queue table does not exist

Some sites have shown issues where Custom tables was enabled, but they were getting server errors because the custom table DB was not present. This is blocking any sync actions from flowing. The idea is to disable custom tables, WPCOM will try to re-enable them, but if not it will use the option tables as a fallback. In any case it should solve the problem and sync actions will start flowing again.
## Proposed changes:


Disables custom table by updating the `custom_queue_table_enabled` setting in case we get an error with the `jetpack_sync_queue`  table doesn't exist. It checks in  `get_item_count` since it is the first query we normally do.


### Other information:


## Jetpack product discussion


p7pQDF-9yY-p2
D146887-code

## Does this pull request change what data or activity we track or use?



## Testing instructions:








Need a test site, with custom tables enabled, which is the default.

First try without the patch.
* Remove the `jetpack_sync_queue` table from the Wordpress DB.
* Create some sync activity (e.g.Create a post)
* See that no sync actions are flowing and the site keeps getting errors about `jetpack_sync_queue` table doesn't exist.

You can get back to normal by manually disabling custom tables.

Now with the patch 
* Remove the `jetpack_sync_queue` table from the Wordpress DB.
* Create some sync activity (e.g.Create a post)
* See that the site disables custom tables. WPCOM will try to re-enable it and re-create the table.
* See that sync actions flow normally.
